README.md

```diff
@@ -251,6 +254,30 @@ Future.node(done => fs.readFile('package.json', 'utf8', done))
 //> "{...}"
 ```
 
+#### loop
+##### `.loop :: (b -> Future a (Iteration b)) -> b -> Future a b`
+
+Stack- and memory-safe asynchronous "recursion" based on [FantasyLand ChainRec][26].
+Calls the given function with the initial value, and expects a Future of an
+[Iteration](#type-signatures). If the Iteration is incomplete (`{done: false}`),
+then the function is called again with the Iteration value until it returns a
+Future of a complete (`{done: true}`) Iteration.
+
+```js
+const Next = x => ({done: false, value: x});
+const Done = x => ({done: true, value: x});
+Future.loop((x => Future.of(x < 1000000 ? Next(x + 1) : Done(x))), 0)
+.fork(console.error, console.log);
+//> 1000000
+```
+
+#### chainRec
+##### `.chainRec :: ((((b -> Next b), (b -> Done b), b) -> Future a (Iteration b)), b) -> Future a b`
+
+Fantasy-Land compliant version of [`loop`](#loop). Passes Next and Done
```

Comment:
This should be "loop", because I intended `loop` to be leading in the eyes of the user (and `chainRec` more like an internal function only exposed for interoperability). Though it seems @rpominov disagrees, I'll need to consider it some more.